recipe-server/client/control_new/components/forms/FormItem.js

```diff
@@ -50,21 +55,40 @@ export default class FormItem extends React.Component {
     initialValue: null,
     name: null,
     rules: null,
+    trimWhitespace: false,
   };
 
+  trimValue(event) {
+    // InputNumber passes the value as the parameter,
+    // but Input passes it via event.target.value.
+    let value = event;
+    if (event && event.target) {
+      value = event.target.value;
+    }
+    return value.trim();
+  }
+
   render() {
+    let { config } = this.props;
```

Comment:
While technically you can `let` component props, I think it's better to consider them constants and not alter their values.

Looking over how the `config` stuff is set up, I think it could be condensed a bit down to just including the `getValueFromEvent` business in the `getFieldDecorator`, ex:

```
 field = form.getFieldDecorator(name, {
         initialValue,
         rules,
         ...config,
        getValueFromEvent: trimWhitespace  ? this.trimValue : config.getValueFromEvent || null,
})(children);
```

We could still include the console error/warning re: multiple `getValue`s, but I don't think it's necessary to alter/set up a new config object.